test/built-ins/TypedArray/prototype/sort/detached-buffer-comparefn.js

```diff
@@ -0,0 +1,37 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-%typedarray%.prototype.sort
+description: Throws a TypeError if comparefn detaches the object buffer
+info: >
+  22.2.3.26 %TypedArray%.prototype.sort ( comparefn )
+
+  When the TypedArray SortCompare abstract operation is called with two
+  arguments x and y, the following steps are taken:
+
+  ...
+  2. If the argument comparefn is not undefined, then
+    a. Let v be ? Call(comparefn, undefined, « x, y »).
+    b. If IsDetachedBuffer(buffer) is true, throw a TypeError exception.
+    ...
+  ...
+includes: [testTypedArray.js, detachArrayBuffer.js]
+---*/
+
+testWithTypedArrayConstructors(function(TA) {
+  var sample = new TA(2);
```

Comment:
With an array length of 2, this test can't tell the difference between these two implementations:

  1. If `compareFunction` detaches the buffer, immediately return an abrupt
     completion (the correct behavior)
  2. Sort the array, and if the buffer is detached at the end of the sorting
     operation, return an abrupt completion (incorrect behavior)

`#2` seems unlikely, but if we're going to have a `calls` variable, we might as well use it!